src/org/lsst/ts/jenkins/components/Csc.groovy

```diff
@@ -93,17 +93,23 @@ def build_salobj_conda(label, concatVersion) {
     """
 }
 
-def upload_conda(name, label) {
+def upload_conda(name, label, arch) {
     // Upload the conda package
     // Takes the name of the package and a label
-    if((name.contains("ts-salobj")) || (name.contains("ts-idl"))) {
-        arch = "noarch"
-    } else {
-        arch = "linux-64"
+    if (arch=="aarch64") {
+        arch_loc = "linux-aarch64"
+    } else if (arch=="noarch") {
+        arch_loc = "noarch"
+    } else if (arch=="linux-64"){
+        arch_loc = "linux-64"
+    }
+    else {
+        currentBuild.result = 'ABORTED'
+        error('Please properly define the arch parameter.')
```

Comment:
This is much better! However, it looks like the arch_loc variable can still be removed. Sorry for my previous confusing comment but if arch can be "noarch", "linux-64" or "linux-aarch64" then there is no need for arch_loc.